dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -265,27 +265,46 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
     std::wstring iconFilePath{};
 
     wil::unique_prop_variant propVariantIcon;
-    // Throw in case of failure, since Icon is mandatory and we don't have a fallback!
-    THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
-    THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt == VT_EMPTY, "You must specify an app icon before calling Register().");
+    // In case there is no icon asset, do not throw. Set a default icon instead.
+    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
 
-    iconFilePath = propVariantIcon.pwszVal;
-
-    // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
-    // since .ico files can have multiple icons in the same file.
-    // NotificationController doesn't seem to support such format, so let it take the first icon by default.
-    auto iteratorForCommaDelimiter = iconFilePath.find_first_of(L",");
-    if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+    if (propVariantIcon.vt == VT_LPWSTR && propVariantIcon.pwszVal != nullptr)
     {
-        iconFilePath.erase(iteratorForCommaDelimiter);
-    }
+        iconFilePath = propVariantIcon.pwszVal;
+
+        // Most icon filepaths from Shell APIs have this format: <filepath>,-<index>,
+        // since .ico files can have multiple icons in the same file.
+        // NotificationController doesn't seem to support such format, so let it take the first icon by default.
+        auto iteratorForCommaDelimiter = iconFilePath.find_first_of(L",");
+        if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+        {
+            iconFilePath.erase(iteratorForCommaDelimiter);
+        }
+
+        auto iteratorForLastBackSlash = iconFilePath.find_last_of(L"\\");
+        auto iteratorForForwardSlash = iconFilePath.find_last_of(L"/");
+        THROW_HR_IF_MSG(ERROR_BAD_PATHNAME, iteratorForLastBackSlash != std::wstring::npos && iteratorForForwardSlash != std::wstring::npos,
```

Comment:
Break this up please to improve readablility. For example, have the first check after 
auto iteratorForLastBackSlash = iconFilePath.find_last_of(L"\\");
And have the second check after:
auto iteratorForForwardSlash = iconFilePath.find_last_of(L"/");

Reply:
Honestly, I'm not sure I understand what this check is about. What path are you considering as invalid and planning on throwing for? Add a comment please.

Reply:
Why can't we simply prevent forward slash and keep it consistent instead of this complex check.

Reply:
To me c:\test\test.png is valid. It only supports backslash. So fail only if forward slash is part of the path.